index.html

```diff
@@ -9142,6 +9161,252 @@
+          to the amount.</p>
+          <table class = "parameters">
+            <tbody>
+              <tr>
+                <th>Parameter</th>
+                <th>Type</th>
+                <th>Nullable</th>
+                <th>Optional</th>
+                <th>Description</th>
+              </tr>
+              <tr>
+                <td class="prmName">amount</td>
+                <td class="prmType"><code>unsigned long</code></td>
+                <td class="prmNullFalse"><span role="img" aria-label=
+                "False">&#10008;</span></td>
+                <td class="prmOptFalse"><span role="img" aria-label=
+                "False">&#10008;</span></td>
+                <td class="prmDesc"></td>
+              </tr>
+              <tr>
+                <td class="prmName">maxBufferedAmount</td>
+                <td class="prmType"><code>unsigned long</code></td>
+                <td class="prmNullFalse"><span role="img" aria-label=
+                "False">&#10008;</span></td>
+                <td class="prmOptTrue"><span role="img" aria-label=
+                "True">&#10004;</span></td>
+                <td class="prmDesc"></td>
+              </tr>
+            </tbody>
+          </table>
+          <div>
+            <em>Return type:</em> <code>Promise&lt;void&gt;</code>
+          </div>
+        </dd>
+        <dt><code>read</code></dt>
+        <dd>
+          <p><dfn>read()</dfn> reads all the data that is in the buffer.
+          <div>
+            <em>No parameters.</em>
+          </div>
+          <div>
+            <em>Return type:</em> <code>Uint8Array</code>
+          </div>
+        </dd>
+        <dt><code>readInto</code></dt>
+        <dd>
+          <p><dfn>readInto()</dfn> fills the array exactly from the read buffer. If that amount
+            is not available, it will fail.
+          <table class = "parameters">
+            <tbody>
+              <tr>
+                <th>Parameter</th>
+                <th>Type</th>
+                <th>Nullable</th>
+                <th>Optional</th>
+                <th>Description</th>
+              </tr>
+              <tr>
+                <td class="prmName">data</td>
+                <td class="prmType"><code>Uint8Array</code></td>
+                <td class="prmNullFalse"><span role="img" aria-label=
+                "False">&#10008;</span></td>
+                <td class="prmOptFalse"><span role="img" aria-label=
+                "False">&#10008;</span></td>
+                <td class="prmDesc"></td>
+              </tr>
+            </tbody>
+          </table>
+          <div>
+            <em>Return type:</em> <code>void</code>
+          </div>
+        </dd>
+        <dt><code>waitForReadableOrFinished</code></dt>
+        <dd>
+          <p><dfn>waitForReadableOrFinished()</dfn> will be resolved when the amount, up to
+          maxBufferedAmount, can be read. If the maxBufferedAmount is not provided, it will
+           be equal to the amount.</p>
```

Comment:
It will be resolved when "amount" is available in the buffer, and the max buffered amount is maxBufferedAmount. In other words, it will keep filling up to maxBufferedAmount, even if it's more than "amount" (it can keep filling even after the promise resolves).